src/Control/ElementFormController.php

```diff
@@ -55,7 +56,19 @@ public function finished()
         $user->finished();
 
         $page = $this->getPage();
-        $controller = Injector::inst()->create($page->getControllerName(), $this->element->getPage());
+
+        if ($page === null) {
+            return null;
+        }
+
+        while (!$page instanceof SiteTree) {
```

Comment:
```suggestion
        while ($page instanceof BaseElement) {
```
Please also remove the `use` statement for SiteTree above.